src/ser/mod.rs

```diff
@@ -189,3 +189,16 @@ where
     let ser = Serializer::new();
     value.serialize(ser)
 }
+
+/// Encode a `T` Serializable into a BSON `Document`.
+pub fn to_document<T: ?Sized>(value: &T) -> Result<Document>
+where
+    T: Serialize,
+{
+    let bson = to_bson(value)?;
+    bson.as_document()
```

Comment:
I think it would be more efficient to just pattern match the value and return it directly instead of cloning it. I know that using combinators is more fun though, but it's worth it to keep things as quick as possible for users :/